policy/modules/services/container.te

```diff
@@ -0,0 +1,720 @@
+
+ps_process_pattern(container_engine_system_domain, container_system_domain)
+allow container_system_domain container_engine_system_domain:fd use;
+allow container_system_domain container_engine_system_domain:fifo_file rw_fifo_file_perms;
+
+create_dirs_pattern(container_engine_system_domain, container_config_t, container_config_t)
+files_etc_filetrans(container_engine_system_domain, container_config_t, dir)
+
+manage_dirs_pattern(container_engine_system_domain, container_log_t, container_log_t)
+manage_files_pattern(container_engine_system_domain, container_log_t, container_log_t)
+logging_log_filetrans(container_engine_system_domain, container_log_t, { dir file })
+
+allow container_engine_system_domain container_var_lib_t:dir { manage_dir_perms relabel_dir_perms watch };
+allow container_engine_system_domain container_var_lib_t:file { manage_file_perms relabel_file_perms exec_file_perms };
+allow container_engine_system_domain container_var_lib_t:blk_file { manage_blk_file_perms relabel_blk_file_perms };
+allow container_engine_system_domain container_var_lib_t:chr_file { manage_chr_file_perms relabel_chr_file_perms };
+allow container_engine_system_domain container_var_lib_t:fifo_file { manage_fifo_file_perms relabel_fifo_file_perms };
+allow container_engine_system_domain container_var_lib_t:lnk_file { manage_lnk_file_perms relabel_lnk_file_perms };
+allow container_engine_system_domain container_var_lib_t:sock_file { manage_sock_file_perms relabel_sock_file_perms };
+files_var_lib_filetrans(container_engine_system_domain, container_var_lib_t, dir)
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, file, "config.env")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, file, "hosts")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, file, "hostname")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, file, "resolv.conf")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, dir, "init")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, dir, "overlay")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, dir, "overlay-images")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, dir, "overlay-layers")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, dir, "overlay2")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, dir, "overlay2-images")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_ro_file_t, dir, "overlay2-layers")
+filetrans_pattern(container_engine_system_domain, container_var_lib_t, container_file_t, dir, "volumes")
```

Comment:
I don't see a "default" filetrans with no name for `container_engine_system_domain` in `container_var_lib_t` dirs.. Shouldn't this be it? (drop the names in this line?)

Reply:
Only a few of the directories in `/var/lib/containers/storage` will be `container_ro_file_t` or `container_file_t`. Everything else will remain `container_var_lib_t`. The listing should look similar to:
```
drwx------. 1 root root system_u:object_r:container_var_lib_t:s0   50 Aug 10 17:36 cache
drwx------. 1 root root system_u:object_r:container_var_lib_t:s0   64 Oct 12 08:25 libpod
drwx------. 1 root root system_u:object_r:container_var_lib_t:s0    0 Mar  3  2021 mounts
drwx--x--x. 1 root root system_u:object_r:container_ro_file_t:s0 2562 Jan 20 16:28 overlay
drwx--x--x. 1 root root system_u:object_r:container_var_lib_t:s0  188 Jan 20 16:28 overlay-containers
drwx------. 1 root root system_u:object_r:container_ro_file_t:s0  428 Jan 20 16:09 overlay-images
drwx------. 1 root root system_u:object_r:container_ro_file_t:s0 2816 Jan 20 16:28 overlay-layers
drwx------. 1 root root system_u:object_r:container_var_lib_t:s0   24 Dec  4 18:30 secrets
-rw-r--r--. 1 root root system_u:object_r:container_var_lib_t:s0   64 Jan 19 14:16 storage.lock
drwx------. 1 root root system_u:object_r:container_var_lib_t:s0    0 Mar  3  2021 tmp
-rw-r--r--. 1 root root system_u:object_r:container_var_lib_t:s0    0 Mar  3  2021 userns.lock
drwx------. 1 root root system_u:object_r:container_var_lib_t:s0    0 Jun 22  2021 vfs
drwx------. 1 root root system_u:object_r:container_var_lib_t:s0   30 Jun 22  2021 vfs-containers
drwx------. 1 root root system_u:object_r:container_var_lib_t:s0   22 Jun 22  2021 vfs-images
drwx------. 1 root root system_u:object_r:container_var_lib_t:s0   22 Jun 22  2021 vfs-layers
drwx--x--x. 1 root root system_u:object_r:container_file_t:s0    2722 Jan 15 18:20 volumes
```

The same should be true of `HOME_DIR/.local/share/containers/storage`.

Reply:
Ok.